lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,277 @@
+# POSSIBILITY OF SUCH DAMAGE.
+""" Utilites to find the relative position of the sun."""
+
+import numpy as np
+import datetime as dt
+
+from improver.utilities.temporal import iris_time_to_datetime
+from improver.utilities.spatial import (
+    lat_lon_determine, transform_grid_to_lat_lon)
+
+
+def solar_declination(day_of_year):
+    """
+    Calculate the Declination for the day of the year.
+
+    Calculation equivalent to the calculation defined in
+    NOAA Earth System Reseach Lab Low Accuracy Equations
+    https://www.esrl.noaa.gov/gmd/grad/solcalc/sollinks.html
+
+    Args:
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+
+    Returns:
+        declination (float):
+            Declination in degrees.North-South
+    """
+    # Declination (degrees):
+    # = -(axial_tilt)*cos(360./orbital_year * day_of_year - solstice_offset)
+    declination = -23.5 * np.cos(np.radians(0.9856 * day_of_year + 9.3))
+    return declination
+
+
+def solar_hour_angle(longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Solar Hour angle for an array of longitudes.
+
+    Calculation equivalent to the calculation defined in
+    NOAA Earth System Reseach Lab Low Accuracy Equations
+    https://www.esrl.noaa.gov/gmd/grad/solcalc/sollinks.html
+
+    Args:
+        longitudes (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0 degrees
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC
+
+    Returns:
+        solar_hour_angle (numpy.array)
+            Hour angle in degrees East-West
+    """
+    if np.min(longitudes) < -180.0 or np.max(longitudes) > 180.0:
+        msg = ('Longitudes must be between -180.0 and 180.0')
+        raise ValueError(msg)
+    thetao = 2*np.pi*day_of_year/365.0
+    eqt = (0.000075 + 0.001868 * np.cos(thetao) -
+           0.032077 * np.sin(thetao) - 0.014615 * np.cos(2*thetao) -
+           0.040849 * np.sin(2*thetao))
+
+    # Longitudinal Correction from the Grenwich Meridian
+    lon_correction = 24.0*longitudes/360.0
+    # Solar time (hours):
+    solar_time = utc_hour + lon_correction + eqt*12/np.pi
+    # Hour angle (degrees):
+    solar_hour_angle = (solar_time - 12.0) * 15.0
+
+    return solar_hour_angle
+
+
+def solar_elevation(latitudes, longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Solar elevation.
+
+    Args:
+        latitudes (float or numpy.array):
+            A single Latitude or array of Latitudes
+            latitudes needs to be between -90.0 and 90.0
+        longitudes (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC in hours
```

Comment:
Invalid day / hour inputs: this function should catch errors raised by "solar_declination" and "solar_hour_angle".